Remove dependencies on the `mozdata` project from ETL

sean-rose opened the pull request:
Having dependencies on things in `mozdata` can cause issues with deployments, as deploying things to `mozdata` is usually a separate secondary step.

---
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

sean-rose commented:
There are a few dependencies on `mozdata` that can't be removed because there's no equivalent in `moz-fx-data-shared-prod`:
- [sql/moz-fx-data-bq-performance/release_criteria/rc_flattened_test_data_v1/query.sql](https://github.com/mozilla/bigquery-etl/blob/a132d631a649fc685b232e48929766b2e2bf15d3/sql/moz-fx-data-bq-performance/release_criteria/rc_flattened_test_data_v1/query.sql#L40)
  - `mozdata.taskclusteretl.perfherder`
- [sql/moz-fx-data-shared-prod/looker_derived/desktop_forecasts_cache/view.sql](https://github.com/mozilla/bigquery-etl/blob/a132d631a649fc685b232e48929766b2e2bf15d3/sql/moz-fx-data-shared-prod/looker_derived/desktop_forecasts_cache/view.sql#L50-L53)
  - `mozdata.analysis.loines_desktop_dau_forecast_2021-01-19`
  - `mozdata.analysis.loines_desktop_new_profiles_forecast_2021-01-19`
- [sql/moz-fx-data-shared-prod/telemetry_derived/active_users_aggregates_v1/query.sql](https://github.com/mozilla/bigquery-etl/blob/a132d631a649fc685b232e48929766b2e2bf15d3/sql/moz-fx-data-shared-prod/telemetry_derived/active_users_aggregates_v1/query.sql#L53)
  - `mozdata.static.csa_gblmkt_languages`

sean-rose commented on dags/bqetl_fog_decision_support.py:
```diff
@@ -55,6 +55,19 @@
+        pool="DATA_ENG_EXTERNALTASKSENSOR",
+    )
+
+    fog_decision_support_v1.set_upstream(wait_for_copy_deduplicate_all)
```
I wasn't expecting changing the `mozdata` dependencies to have any effect on the Airflow task dependencies, since any views deployed to both `moz-fx-data-shared-prod` and `mozdata` should be exactly the same.

akkomar replied:
So automation missed these dependencies before?

sean-rose replied:
Indeed.  I tracked down the bug responsible in the DAG generation code and filed a separate fix for that: #3504

akkomar reviewed (commented):
LGTM, but shouldn't we at least document that `mozdata` shouldn't be referenced in queries here? Or ideally add a test covering that?

sean-rose commented:
> @akkomar 
> LGTM, but shouldn't we at least document that `mozdata` shouldn't be referenced in queries here? Or ideally add a test covering that?

I've added a commit to this PR which adds notes to the [recommended practices](https://mozilla.github.io/bigquery-etl/reference/recommended_practices/) doc saying we shouldn't "refer to views in the `mozdata` project which are duplicates of views in another project."

Adding tests/CI to check for `mozdata` references is more complicated than I'd like to do for this PR, especially since we can't forbid all `mozdata` references as there are some cases where there isn't an equivalent resource in another project.  However, I added a comment about this to [bug 1760239](https://bugzilla.mozilla.org/show_bug.cgi?id=1760239#c4) "Additional CI Checks for private-bigquery-etl / bigquery-etl".
